pkg/operation/botanist/health_check.go

```diff
@@ -187,9 +182,22 @@ func (b *HealthChecker) checkRequiredDaemonSets(condition gardencorev1beta1.Cond
 		actualNames.Insert(object.Name)
 	}
 
-	if missingNames := requiredNames.Difference(actualNames); missingNames.Len() != 0 {
-		c := b.FailedCondition(condition, "DaemonSetMissing", fmt.Sprintf("Missing required daemon sets: %v", missingNames.List()))
-		return &c
+	return b.checkRequiredResourceNames(condition, requiredNames, actualNames, "DaemonSetMissing", "Missing required daemon sets")
+}
+
+var relevantConditions = map[resourcesv1alpha1.ConditionType]struct{}{
+	resourcesv1alpha1.ResourcesApplied: struct{}{},
+	resourcesv1alpha1.ResourcesHealthy: struct{}{},
+}
+
+// CheckManagedResource checks the conditions of the given managed resource and reflects the state in the returned condition.
+func (b *HealthChecker) CheckManagedResource(condition gardencorev1beta1.Condition, mr *resourcesv1alpha1.ManagedResource) *gardencorev1beta1.Condition {
+	for _, cond := range mr.Status.Conditions {
+		_, relevant := relevantConditions[cond.Type]
+		if relevant && cond.Status == resourcesv1alpha1.ConditionFalse {
```

Comment:
Shouldn't the condition also fail if the relevant type wasn't found?

Reply:
Yes, good idea.